tests/test_bot.py

```diff
@@ -588,24 +583,27 @@ def test_send_close_date_default_tz(self, tz_bot, super_group_id):
         aware_close_date = dtm.datetime.now(tz=tz_bot.defaults.tzinfo) + dtm.timedelta(seconds=5)
         close_date = aware_close_date.replace(tzinfo=None)
 
-        message = tz_bot.send_poll(
+        msg = tz_bot.send_poll(
             chat_id=super_group_id,
             question=question,
             options=answers,
             close_date=close_date,
             timeout=60,
         )
-        assert message.poll.close_date == aware_close_date.replace(microsecond=0)
+        # Sometimes there can be a few seconds delay, so don't let the test fail due to that-
+        if 1 <= abs(msg.poll.close_date.second - aware_close_date.second) < 5:
+            msg.poll.close_date = msg.poll.close_date.replace(second=aware_close_date.second)
+        assert msg.poll.close_date == aware_close_date.replace(microsecond=0)
```

Comment:
maybe just check that the timedelta `msg.poll.close_data - aware_close_date` is small?

Reply:
oh we can't do that since they are both in different timezones, hence they can have different minutes/hours. 

what I learned from this is that two datetimes can be equal even if their timezones are not equal.

Reply:
As long as the *both* are aware, that shouldn't be a problem - since they are aware, you can convert them both to a common timezone. `(msg.poll.close_date - aware_close_date).total_seconds() <= 5` works fine for me …